share/completions/iwctl.fish

```diff
@@ -0,0 +1,59 @@
+set adpaters '(iwctl adpater list | tail -n -2 | awk \'{print $1}\')'
+set devices '(iwctl device list | tail -n -2 | awk \'{print $1}\')'
+set dpp '(iwctl dpp list | tail -n -2 | awk \'{print $1}\')'
+set knownnetworks '(iwctl known-networks list | tail -n -2 | awk \'{print $1}\')'
+set stations '(iwctl station list | tail -n -2 | awk \'{print $1}\')'
+set wsc '(iwctl wsc list | tail -n -2 | awk \'{print $1}\')'
+set net '(iwctl station wlan0 get-networks | tail -n -2 | awk \'{print $2}\')'
+
+complete -f iwctl
+
+# First argument
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 1" \
+   -a "ad-hoc adapter ap device dpp help known-networks station version wsc"
+
+# Second argument
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 2; and __fish_seen_subcommand_from adapter" \
+    -xa "list $adapters"
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 2; and __fish_seen_subcommand_from ap" \
+    -xa "list $ap"
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 2; and __fish_seen_subcommand_from dpp" \
+    -xa "list $dpp"
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 2; and __fish_seen_subcommand_from ad-hoc" \
+    -xa "list $adhoc"
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 2; and __fish_seen_subcommand_from device" \
+    -xa "list $devices"
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 2; and __fish_seen_subcommand_from known-networks" \
+    -xa "list $knownnetworks"
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 2; and __fish_seen_subcommand_from station" \
+    -xa "list $stations"
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 2; and __fish_seen_subcommand_from wsc" \
+    -xa "list $wsc"
+
+# Third arguments
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 3; and __fish_seen_subcommand_from device" \
+    -xa "set-property show"
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 3; and __fish_seen_subcommand_from dpp" \
+    -xa "start-configurator start-enrollee stop"
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 3; and __fish_seen_subcommand_from known-networks" \
+    -xa "forget set-property show"
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 3; and __fish_seen_subcommand_from wsc" \
+    -xa "cancel push-button start-pin start-user-pin"
+
+# Forth arguement
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 4; and __fish_seen_subcommand_from device set-property" \
+    -xa  "Powered Mode"
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 4; and __fish_seen_subcommand_from known-networks set-property" \
+    -xa  "AutoConnect"
+complete -c iwctl -n "test (__fish_number_of_cmd_args_wo_opts) = 4; and __fish_seen_subcommand_from station connect" \
+    -xa  '(iwctl station wlan0 get-networks | tail -n -2 | awk \'{print $2}\')'
```

Comment:
This prints "psk" for me. Is that what you want?

Is the output here maybe not stable?

Here's what it looks like for me (with network names replaced):

```
                               Available networks                              
--------------------------------------------------------------------------------
    Network name                    Security          Signal
--------------------------------------------------------------------------------
  > currentnetwork                  psk               ****  
    pskdnetwork                     psk               ****
    opennetwork                     open              ****
```

Reply:
For me, it's `print 3` that prints `psk`.

> Is the output here maybe not stable?

I guess

Reply:
Ooof.

>For me, it's print 3 that prints psk.

Does your output look different than what I showed you above, e.g. are the columns swapped, or is this a case of whitespace being different (which we could probably work around).

Because if the columns are swapped, that's terrible news, because it makes this output unusable, which kind of makes the entire script unmergeable.

Reply:
I guess, this is because I only have one network

This is my output
```
                               Available networks
--------------------------------------------------------------------------------
    Network name                    Security          Signal
--------------------------------------------------------------------------------
  > FTTH-6DBD                       psk               ****
```

Reply:
Aaaaah, okay, I see what it is:

The selected network line starts with `  > `, while unselected networks start with `    `. awk will collapse the latter because it's whitespace. In the former the `>` is the first field!

So: `awk '{print $2}'` is too simple here. You probably want a regular expression.

E.g.

```fish
string replace -rf '^[\s>]*(\S+)\s*.*' '$1'
```

And the `>` is even colored, so you have that `\e[SOMENUMBERm` to deal with.

The full line would be something like

```fish
iwctl station wlan0 get-networks | tail -n +5 | string replace -ra '\e\[[\d;]+m' '' | string replace -rf '^[\s>]*(\S+)\s*.*' '$1'
```

(note the `tail -n +5`. For me that's necessary because the first four lines are header guff)